controllers/finalizers.go

```diff
@@ -94,20 +95,29 @@ func finalizeSpecialResource(r *SpecialResourceReconciler) error {
 		key := client.ObjectKeyFromObject(&ns)
 
 		err := clients.Interface.Get(context.TODO(), key, &ns)
-		if apierrors.IsNotFound(err) {
-			log.Info("Successfully finalized (IsNotFound)", "SpecialResource:", r.specialresource.Name)
-			return nil
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Info("Successfully finalized (Namespace IsNotFound)", "SpecialResource:", r.specialresource.Name)
+				return nil
+			} else {
+				log.Error(err, "Failed to get namespace", "namespace", r.specialresource.Spec.Namespace, "SpecialResource", r.specialresource.Name)
+				return err
+			}
 		}
 
 		for _, owner := range ns.GetOwnerReferences() {
 			if owner.Kind == "SpecialResource" {
 				log.Info("Namespaces is owned by SpecialResource deleting")
 				err = clients.Interface.Delete(context.TODO(), &ns)
-				if !apierrors.IsNotFound(err) {
-					warn.OnError(err)
+				if err != nil {
+					log.Error(err, "Failed to delete namespace", "namespace", r.specialresource.Spec.Namespace)
+					return err
```

Comment:
This changes the behaviour of the code - now, an already deleted namespace will cause an error

Reply:
We are checking if the namespace exists in 99, and if it was already deleted we return a success

Reply:
Got it, thanks. Now should we `return` at the end this `if` block`? If the `ownerReference` was specified twice, then we will be trying to delete the namespace twice...
LGTM otherwise.

Reply:
I don't see how OwnerReference can be specified twice with the same values ( it is being managed by the controller-runtime code, i think),but even if it does, the second call for deletion will fail, and reconcile will be rescheduled, and on the next loop we will return success, since Namespace will be non-existent

Reply:
Fair enough.

Reply:
lgtm?